.github/workflows/performance.yml

```diff
@@ -0,0 +1,79 @@
+name: "Compare performance"
+
+on:
+  workflow_call:
+
+env:
+  GIT_CONFIG_GLOBAL: "/root/.gitconfig" # fix path in container (https://github.com/actions/runner/issues/2033)
+
+jobs:
+  compare-performance:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: "ubuntu-latest"
+    container:
+      image: "ghcr.io/tarantool/sdvg-ci:0.0.2"
+    permissions:
+      pull-requests: write
+      packages: read
+
+    steps:
+      - uses: "actions/checkout@v4"
+
+      - name: "Install requirements"
+        run: "make bench/prepare"
+
+      - name: "Try fetch cached benchmark result from master branch"
+        id: "cache-benchmark"
+        uses: "actions/cache/restore@v4"
+        with:
+          key: "benchmark-master"
+          path: "benchmark-master.txt"
+
+      - name: "Fetch master branch"
+        if: ${{ steps.cache-benchmark.outputs.cache-hit != 'true' }}
+        uses: "actions/checkout@v4"
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          path: "master_branch"
+
+      - name: "Create master benchmark"
+        if: ${{ steps.cache-benchmark.outputs.cache-hit != 'true' }}
+        working-directory: "master_branch"
+        run: make test/performance > ../benchmark-master.txt
+
+      - name: "Save benchmark-master.txt to cache"
+        if: ${{ steps.cache-benchmark.outputs.cache-hit != 'true' }}
+        uses: "actions/cache/save@v4"
+        with:
+          key: "benchmark-master"
+          path: "benchmark-master.txt"
+
+      - name: "Run benchmarks on PR branch"
+        run: make test/performance > benchmark-pr.txt
+
+      - name: "Compare results"
+        run: make bench/compare > performance-report.md
+
+      - uses: "mshick/add-pr-comment@v2"
```

Comment:
It might be better to do this not for every commit, but for approval or manual launch. Also it will be useful to print comparison results in logs on every commit

The second option is to remove old benchmark results and add new ones with each commit.

Reply:
I didn't understand about the second option.

Now, if there is a benchmark-master in the cache, then I use it and update the cache with the same file.

If you need to update the file benchmark-master.txt every time  then the meaning of the cache is lost.

The file benchmark-master.txt updated:

- when there was a merger in the master
- when the file has expired after 7 days and it is created in actions for PR (any PR which first encountered the cache missing)

Reply:
I have now found out that the cache can be overwritten using the same key through separate manipulations.

Reply:
I said about PR comment, not about caches

Reply:
yes, the comment is being updated